spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAddFilesProcedure.java

```diff
@@ -607,6 +607,40 @@ public void addFilteredPartitionsToPartitionedWithNullValueFilteringOnDept() {
         sql("SELECT id, name, dept, subdept FROM %s ORDER BY id", tableName));
   }
 
+  @TestTemplate
+  public void addFileTableInvalidPartitionSpecVersion() {
+    createPartitionedFileTable("parquet");
+    createIcebergTable(
+        "id Integer, name String, dept String, subdept String", "PARTITIONED BY (id, dept)");
+
+    assertThatThrownBy(
+            () ->
+                sql(
+                    "CALL %s.system.add_files('%s', '`parquet`.`%s`', map('id', 1), true, 1, 1)",
+                    catalogName, tableName, fileTableDir.getAbsolutePath()))
+        .isInstanceOf(IllegalArgumentException.class)
+        .hasMessage("Invalid partition spec version: 1");
+  }
+
+  @TestTemplate
+  public void addFileTableWithPartitionSpecVersion() {
+    createPartitionedFileTable("parquet");
+    createIcebergTable(
+        "id Integer, name String, dept String, subdept String", "PARTITIONED BY (id)");
+    sql("ALTER TABLE %s ADD PARTITION FIELD dept", tableName);
+
+    List<Object[]> result =
+        sql(
+            "CALL %s.system.add_files(table => '%s', source_table => '`parquet`.`%s`', partition_spec_version => 0)",
+            catalogName, tableName, fileTableDir.getAbsolutePath());
```

Comment:
This test fails if we don't add `partition_spec_version => 0` which is why adding this feature

Reply:
N/A with latest change